src/main/java/io/github/techstreet/dfscript/event/RecieveSoundEvent.java

```diff
@@ -1,18 +1,31 @@
 package io.github.techstreet.dfscript.event;
 
 import io.github.techstreet.dfscript.event.system.CancellableEvent;
+import net.minecraft.client.sound.SoundInstance;
+import net.minecraft.network.packet.s2c.play.PlaySoundFromEntityS2CPacket;
+import net.minecraft.network.packet.s2c.play.PlaySoundIdS2CPacket;
 import net.minecraft.network.packet.s2c.play.PlaySoundS2CPacket;
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.util.Identifier;
 
 public class RecieveSoundEvent implements CancellableEvent {
     private boolean cancelled = false;
-    private final PlaySoundS2CPacket packet;
 
-    public RecieveSoundEvent(PlaySoundS2CPacket packet) {
-        this.packet = packet;
+    private final SoundInstance sound;
+
+    public RecieveSoundEvent(SoundInstance sound) {
+        this.sound = sound;
+    }
+
+    public Identifier getSoundId() {
+        return sound.getId();
+    }
+    public float getVolume() {
```

Comment:
`Cannot invoke "net.minecraft.client.sound.Sound.getVolume()" because "this.sound" is null`
Also with getPitch